MUXSDKStats/MUXSDKStats/MUXSDKPlayerBinding.m

```diff
@@ -82,12 +82,97 @@ - (void)attachAVPlayer:(AVPlayer *)player {
+
+                NSString *domain = [self getHostName:event.URI];
+                MUXSDKBandwidthMetricData *loadData = [[MUXSDKBandwidthMetricData alloc] init];
+                loadData.requestType = @"ManifestOrMedia";
+                loadData.requestStart = [NSNumber numberWithLong: _lastTransferDuration * 1000];
+                loadData.requestResponseStart = [NSNumber numberWithLong: _lastTransferDuration * 1000];
+                loadData.requestResponseEnd = [NSNumber numberWithLong: event.transferDuration * 1000];
+                loadData.requestBytesLoaded = [NSNumber numberWithLong: event.numberOfBytesTransferred - _lastTransferredBytes];
+                loadData.requestResponseHeaders = nil;
+                loadData.requestHostName = (domain == nil) ? event.serverAddress : domain;
+                loadData.requestCurrentLevel = nil;
+                loadData.requestMediaStartTime = nil;
+                loadData.requestMediaDuration = nil;
+                loadData.requestVideoWidth = nil;
+                loadData.requestVideoHeight = nil;
+                loadData.requestRenditionLists = nil;
+                [self dispatchBandwidthMetric:loadData];
+
+                // manual bitrate calculation; it includes request latency, which is defined as delta between request sent and 1st byte of recsponse received.
+                // double currBitrate = transferredBytes * 8 / transferDuration;
+                // if (_minBitrate == 0 || _minBitrate > currBitrate) {
+                //    _minBitrate = currBitrate;
+                // }
+                // _totalTime += transferDuration;
+                // _totalBytes += transferredBytes;
+                // double avgBitRate = _totalBytes * 8 / _totalTime;
+                // NSLog(@"Bwm: min %f, avg %f", _minBitrate, avgBitRate);
+                _lastTransferredBytes = event.numberOfBytesTransferred;
+                _lastTransferDuration = event.transferDuration;
+            }
+            _lastMediaRequestBytes = event.numberOfBytesTransferred;
+            _lastMediaRequest = event.numberOfMediaRequests;
+        }
+
+        AVPlayerItemErrorLog *error = _player.currentItem.errorLog;
+        if (error != nil && error.events.count > 0) {
+            // https://developer.apple.com/documentation/avfoundation/avplayeritemerrorlogevent?language=objc
+            if (_lastErrorLogEventCount < error.events.count) {
+                AVPlayerItemErrorLogEvent *errorEvent = error.events[error.events.count - 1];
+                NSString *domain = [self getHostName:errorEvent.URI];
+                MUXSDKBandwidthMetricData *loadData = [[MUXSDKBandwidthMetricData alloc] init];
+                loadData.requestError = errorEvent.errorDomain;
+                loadData.requestType = @"ManifestOrMedia";
+                loadData.requestUrl = errorEvent.URI;
```

Comment:
Can we pass the whole URL in for the non-error case, too, and put the logic for getting the hostname inside Core?

Reply:
extracting hostname requires OS dependent API (see inside the getHostName function at line 94 and line 95). i imagine the core will be transpiled so that i try to avoid OS dependent APIs as much as possible. hence, I still prefer to put the hostname setting in the integration level

Reply:
Ahh I see that now, good call.